pkg/kind/kong_proxy.go

```diff
@@ -0,0 +1,172 @@
+package kind
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/kong/kubernetes-testing-framework/pkg/generators/k8s"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	kubeinformers "k8s.io/client-go/informers"
+	"k8s.io/client-go/tools/cache"
+)
+
+// -----------------------------------------------------------------------------
+// Kong Proxy Cluster - Events
+// -----------------------------------------------------------------------------
+
+// ProxyReadinessEvent indicates the result of exposing the Kong proxy service in a Cluster
+type ProxyReadinessEvent struct {
+	URL *url.URL
+	Err error
+}
+
+// -----------------------------------------------------------------------------
+// Kong Proxy Cluster - Informers
+// -----------------------------------------------------------------------------
+
+// startProxyInformer provides a channel indicates when the proxy server is fully functional and accessible
+// by providing the *url.URL by which to access it. The channel will produce a nil value on failure.
+func (c *kongProxyCluster) startProxyInformer(ctx context.Context) (ready chan ProxyReadinessEvent) {
+	ready = make(chan ProxyReadinessEvent)
+
+	// we need to wait for the Kong proxy deployment
+	deployment := c.startDeploymentInformer(ctx)
+
+	go func() {
+		select {
+		case d := <-deployment:
+			// start the service informer to watch updates to the service we're about to create
+			go c.startServiceInformer(ctx, types.NamespacedName{Namespace: d.Namespace, Name: d.Name}, ready)
+
+			// expose the deployment
+			_, err := c.exposeProxyDeployment(ctx, d)
+			if err != nil {
+				ready <- ProxyReadinessEvent{Err: err}
+				close(ready)
+				return
+			}
+		case <-time.After(time.Minute * 3): // TODO: vars
```

Comment:
Would it make sense to use a context with timeout instead of `time.After`?

I mean replacing
```go
case <-time.After(time.Minute * 3):
```
with
```go
case <-ctx.Done():
```
and running `startProxyInformer()` with a timeout-enabled context.

The benefit would be that the timeout would propagate to all operations carried out in `startProxyInformer`.

Reply:
Yes, that would be better. I'll make this change :+1:

Reply:
2324c54f905e551c16eaed5d6f1a0d2e41742613 takes care of this :thinking: